node/src/components/block_synchronizer/deploy_acquisition.rs

```diff
@@ -52,19 +54,19 @@ impl DeployAcquisition {
     ) -> Result<(), Error> {
         let new_acquisition = match self {
             DeployAcquisition::ByHash(acquisition) => {
-                let mut new_deploy_ids = vec![];
+                let mut new_deploy_ids = HashMap::new();
                 for ((deploy_hash, deploy_state), approvals_hash) in acquisition
                     .inner
-                    .drain(..)
+                    .drain()
```

Comment:
Won't that zip them in the wrong order, since `acquisition.inner` is now a `HashMap`?